### PR TITLE
MBS-11465 runner execution timeout

### PR DESCRIPTION
**Pull request by RuslanMingaliev**

### CODE REVIEW

File: subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt [hunk 1]
```diff
@@ -129,7 +129,7 @@ internal class DeviceWorker(
         stateWorker.clearPackages(currentState)
     }.map { intendedState }
 
-    private fun executeAction(action: InstrumentationTestRunAction): Result<DeviceTestCaseRun> {
+    private suspend fun executeAction(action: InstrumentationTestRunAction): Result<DeviceTestCaseRun> {
```
Comment (sergioboi) [inline subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt:132]: Why do we need the suspend there?

File: subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/device/stub/StubDevice.kt [hunk 2]
```diff
@@ -67,13 +70,16 @@ public open class StubDevice(
         return result
     }
 
-    override fun runIsolatedTest(
+    override suspend fun runIsolatedTest(
         action: InstrumentationTestRunAction,
         outputDir: File
     ): DeviceTestCaseRun {
+        // simulating test execution
```
Comment (sergioboi) [inline subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/device/stub/StubDevice.kt]: Propose to delete this obvious comment

File: subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt (hunk 1 above)
Comment (sergioboi) [reply subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt:132]: Understand

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt [hunk 3]
```diff
@@ -17,7 +17,11 @@ public abstract class InstrumentationConfiguration(public val name: String) {
 
     public var kubernetesNamespace: String = "default"
 
-    public var timeoutInSeconds: Long = TimeUnit.MINUTES.toSeconds(100)
+    public var timeoutInSeconds: Long = 120L // TODO: remove after MBS-11465
```
Comment (sergioboi) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt:21]: Add @Deprecated

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt [hunk 4]
```diff
@@ -284,7 +284,8 @@ public class InstrumentationTestsPlugin : Plugin<Project> {
             kubernetesNamespace = configuration.kubernetesNamespace,
             targets = getTargets(configuration, mergedInstrumentationParameters),
             enableDeviceDebug = configuration.enableDeviceDebug,
-            timeoutInSeconds = configuration.timeoutInSeconds,
+            testRunnerExecutionTimeout = configuration.testRunnerExecutionTimeout,
+            gradleTaskTimeout = configuration.gradleTaskTimeout,
```
Comment (sergioboi) [inline subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: `instrumentationTaskTimeout`
gradleTask is too general description

File: subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt [hunk 5]
```diff
@@ -65,6 +69,8 @@ internal class RunnerIntegrationTest {
         minimumFailedCount = 0
     )
     private val loggerFactory = StubLoggerFactory
+    private val <E> Channel<E>.isClosedForSendAndReceive: Boolean
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt]: Let's move it to place where it's used

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt [hunk 6]
```diff
@@ -30,77 +32,81 @@ internal class TestRunnerImpl(
...
-                state.cancel()
-                devicesProvider.releaseDevices()
+                    testSuiteListener.onTestSuiteFinished()
+                    logger.info("Test run end successfully")
+                    Result.Success(result)
+                } catch (e: Throwable) {
+                    logger.critical("Test run end with error", e)
+                    Result.Failure(e)
+                } finally {
+                    logger.debug("Test run finally block called")
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt]: Delete

**Review by sergioboi — CHANGES_REQUESTED** [review]

File: subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/device/stub/StubDevice.kt (hunk 2 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/service/src/main/kotlin/com/avito/runner/service/worker/device/stub/StubDevice.kt]: ✅

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt (hunk 3 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt:21]: ✅

File: subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt (hunk 4 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/InstrumentationTestsPlugin.kt]: ✅

File: subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt (hunk 5 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt]: ✅

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt (hunk 6 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt]: ✅

File: subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt (hunk 5 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/client/src/test/kotlin/com/avito/runner/scheduler/runner/RunnerIntegrationTest.kt]: Moved this extension to `:common:coroutines-extension`. It seems to be the correct place for this property.

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt [hunk 7]
```diff
@@ -30,77 +32,80 @@ internal class TestRunnerImpl(
...
-                logger.critical("Test run end with error", e)
-                Result.Failure(e)
-            } finally {
-                deviceWorkerPool.stop()
-                state.cancel()
-                devicesProvider.releaseDevices()
+                    testSuiteListener.onTestSuiteFinished()
+                    logger.info("Test run end successfully")
+                    Result.Success(result)
+                } catch (e: Throwable) {
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt:105]: Will we get the TimeoutException there? If will maybe make the different log message

**Review by sergioboi — APPROVED** [review]

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt (hunk 7 above)
Comment (RuslanMingaliev) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt:105]: Yes, we will. Changed the message for timeout

Comment (sergioboi) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt:105]: Expect to see multiple catch blocks but you decide do you want to rewrite this code again

Comment (RuslanMingaliev) [reply subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt:105]: ✅

**Pull request merged**